packages/datadog-plugin-confluentinc-kafka-javascript/test/index.spec.js

```diff
@@ -332,23 +341,59 @@ describe('Plugin', () => {
           })
 
           describe('consumer', () => {
-            beforeEach(() => {
+            beforeEach(async () => {
               nativeConsumer = new Consumer({
                 'bootstrap.servers': '127.0.0.1:9092',
-                'group.id': 'test-group-native'
+                'group.id': 'test-group'
               })
 
-              nativeConsumer.on('ready', () => {
-                nativeConsumer.subscribe([testTopic])
+              await new Promise((resolve, reject) => {
+                nativeConsumer.connect({}, (err) => {
+                  if (err) {
+                    return reject(err)
+                  }
+                  resolve()
+                })
               })
-
-              nativeConsumer.connect()
             })
 
-            afterEach(() => {
-              nativeConsumer.disconnect()
+            afterEach(async () => {
+              await nativeConsumer.unsubscribe()
+              await new Promise((resolve, reject) => {
+                nativeConsumer.disconnect((err) => {
+                  if (err) {
+                    return reject(err)
+                  }
+                  resolve()
+                })
+              })
             })
 
+            function consume (consumer, producer, topic, message) {
+              return new Promise((resolve, reject) => {
+                function doConsume () {
```

Comment:
Is there a reasonable limit to how many times we expect this function to be called before we consider it a failure?

If so, it would be more dev friendly if it threw an error when that happened, instead of waiting for the test to time out (it might not be obvious from a time out test, why it timed out).

Reply:
I added a timeout to this method which will throw after 9.5 seconds (since the test times out after 10). Now itll throw an error saying the consumer failed to consume a message.